### PR TITLE
Merge master + fix unit tests on node >= 11

### PR DESCRIPTION
**Pull request by Kehrlann**
A PR was merged onto master but was never merged back onto dev. This fixes it.

Also, a unit test was broken on node 11 & 12, due to an update in their caching strategy. Calling the `reset()` method now clears the `require.cache` for aliased paths.

For reference, here is the module loading strategy in Node:
- Node 10: https://github.com/nodejs/node/blob/f612a714fa5df59e3e766c70c2d07f51bff5e10f/lib/internal/modules/cjs/loader.js#L503-L508
- Node >= 11: https://github.com/nodejs/node/blob/f607e169612ff60a00f456f3c698a6b41792a8f5/lib/internal/modules/cjs/loader.js#L572-L591

I'll let you review commit 0f41c9f @ilearnio

### CODE REVIEW

**Comment by Kehrlann** [comment]
@ilearnio sorry to bump it but can you take a look ? I can't self-approve, and #51 and #54 are blocked on this PR.

Also, it seems having a dev branch adds more confusion than clarity. Maybe we should resolve to using just master. We could handle releases with tags. Thoughts ?

File: index.js [hunk 1]
```diff
@@ -116,6 +116,15 @@ function reset () {
   // Reset all changes in paths caused by addPath function
   modulePaths.forEach(function (path) {
     removePathHelper(path, require.main.paths)
+
+    // Delete from require.cache if the module has been required before.
+    // This is required for node >= 11
```
Comment (ilearnio) [inline index.js]: @Kehrlann does this mean it will re-require the module each time it is been required?

File: test/specs.js [hunk 2]
```diff
@@ -78,25 +81,76 @@ describe('module-alias', function () {
...
 
-    var src, foo, baz, some, someModule
-    try {
-      src = require('@src/foo')
-      foo = require('@foo')
-      baz = require('@bar/baz')
-      some = require('some/foo')
-      someModule = require('some-module')
-    } catch (e) {}
+    describe('with process.cwd()', function () {
```
Comment (ilearnio) [inline test/specs.js]:
```suggestion
    context('when base working directory is process.cwd()', function () {
```

File: test/specs.js [hunk 3]
```diff
@@ -78,25 +81,76 @@ describe('module-alias', function () {
...
+
+      it('should import default settings from process.cwd()/package.json', function () {
+        process.chdir(path.join(__dirname, 'src'))
+        moduleAlias()
+
+        expectAliasesToBeImported()
+      })
+    })
+
+    describe('by looking up __dirname/../../', function () {
```
Comment (ilearnio) [inline test/specs.js]:
```suggestion
    context('when module-alias package is nested (looking up __dirname/../../)', function () {
```

**Comment by ilearnio** [comment]
@Kehrlann To me it seems that dev branch makes sense because before merging into master we can test if all is working well in dev. It's a good practice to have in master only what is the latest published version

**Comment by ilearnio** [comment]
Just need to keep master up to date with dev. If some PR merged to dev, we should test and merge it to master if it's good. But that was my fault that I didn't do it last time

**Comment by ilearnio** [comment]
I've synchronized the branches

File: index.js (hunk 1 above)
Comment (Kehrlann) [reply index.js]: Only when calling reset(). When you reset, you don't want to find the aliased modules.

This fixes the tests on node 11 and 12.

**Comment by Kehrlann** [comment]
dev branch - works for me, let's keep it. Thanks for synchronizing !

I've also added explicit testing on all node versions between 8 and 12. It might not be necessary, what I wanted to test was node 10 and (11 or 12), which have different import behavior. Thoughts ?

**Comment by ilearnio** [comment]
Looks good to me. More is not less 😉

File: index.js (hunk 1 above)
Comment (ilearnio) [reply index.js]: Oh, I see. Awesome! Approving

**Pull request merged**